File: OCH_Tao/ABC/B/257.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const [N, K,] = input[0].split(" ").map(x => parseInt(x, 10));
let A = input[1].split(" ").map(x => parseInt(x, 10));
const L = input[2].split(" ").map(x => parseInt(x, 10));
for (const i of L) {
  if (A[i - 1] === N) {
    continue;
  } else if (i === K) {
    A[i - 1]++;
  } else if (A[i - 1] < A[i] - 1) {
    A[i - 1]++;
  }
}
console.log(...A);

File: OCH_Tao/ABC/B/257.py
def fn(x):
  return [i for i,j in enumerate(x) if j==True]
N,K,Q = map(int,input().split())
A = set(list(map(int,input().split())))
L = list(map(int,input().split()))
X = [i in A for i in range(1,N+1)]
for i in L:
  tmp = fn(X)[i-1]
  if tmp==N-1:
    continue
  else:
    if not X[tmp+1]:
      X[tmp]=False
      X[tmp+1]=True
print(*map(lambda x:x+1,fn(X)),sep=" ")